lib/web_ui/lib/src/engine/canvaskit/texture.dart

```diff
@@ -0,0 +1,97 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:ui/ui.dart' as ui;
+
+import '../texture.dart';
+import 'canvaskit_api.dart';
+import 'image.dart';
+import 'layer.dart';
+import 'painting.dart';
+
+class CkTexture {
+  CkTexture(this.source);
+
+  final Object source;
```

Comment:
I'm wondering if accepting all possible texture sources is future-proof enough. In particular, Skwasm will move to a multi-threaded rendering model where GPU work will be done through an `OffscreenCanvas` in a web worker. I'm not sure if in that model we can use `<img>` elements as texture sources since there's no DOM in web workers. Maybe there's a way to first convert `<img>` to something that's transferable, such as `VideoFrame`, and then send it to the worker? cc @eyebrowsoffire for guidance.

Reply:
The issue you bring up is something that I'm going to have to work out regardless of this change, and I haven't gotten to the point with the multi-threaded rendering where I have to decide how I'm going to solve it, but I think this should be fine. A `VideoFrame` can be constructed from an `HTMLImageElement` if we end up going that route: https://developer.mozilla.org/en-US/docs/Web/API/VideoFrame/VideoFrame